index.js

```diff
@@ -0,0 +1,90 @@
+const URL = 'https://trektravel.herokuapp.com/trips';
+
+function reportStatus(message) {
+  $('#status-message').html(message);
+}
+
+let trips = {};
+
+function loadTrips() {
+  reportStatus('Loading trips...');
+
+  const tripList = $('#trip-list');
+  tripList.empty();
+
+  axios.get(URL)
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      response.data.forEach((trip) => {
+        const listItem = $(`<li>${trip.name}</li>`).appendTo(tripList);
+        listItem.click(() => {
+          clickTrip(trip.name);
+        })
+        trips[trip.name] = trip;
+      });
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      console.log(error);
+    });
+}
+
+function clickTrip(tripName) {
+  const trip = trips[tripName];
+  const details = $('#details');
+  details.show();
+  details.empty();
+  details.append(`<p><b>Name</b>: ${trip.name}</p>`);
+  details.append(`<p>Continent: ${trip.continent}</p>`);
+  details.append(`<p>Category: ${trip.category}</p>`);
+  details.append(`<p>Weeks: ${trip.weeks}</p>`);
+  details.append(`<p>Cost: $${trip.cost}</p>`);
+  details.append(`<p>About:</p>`);
+  details.append(`<p>This is the trip of your dreams!</p>`);
+
+  const form = $('#registration');
+  form.show();
+  $('#registration-submit').click((event) => {
```

Comment:
When reserving a second trip, your site makes two POST requests. To reproduce:
1. Open page
1. Open chrome dev tools to the "Network" tab
1. Click See All Trips
1. Click on the first trip (Cairo to Zanzibar)
1. Make a reservation and submit
1. Click on the second trip (Everest Base Camp Trek)
1. Make a reservation and submit
1. Observe that on submit, 2 POST requests were made to trek API's /reservations endpoint

The reason for this is that you reuse the same reservation form for each trip, but re-attach the `submit` handler every time you show details for a trip.